packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts

```diff
@@ -464,19 +465,78 @@ export class OdspDocumentStorageService implements IDocumentStorageService {
+            let stToRespEnd; // responseEnd - startTime
+            const spReqDuration = response.headers.get("sprequestduration");
+
+            if (performance) {
+                const resources1 = performance.getEntriesByType("resource");
+                for (let i = resources1.length - 1; i > 0; i--) {
+                    const indResTime = resources1[i] as PerformanceResourceTiming;
+                    const resource_name = indResTime.name;
+                    const resource_initiatortype = indResTime.initiatorType;
+                    if ((resource_initiatortype.localeCompare("fetch") === 0) && (resource_name.localeCompare(url) === 0)) {
+                        isResourceDataAvailable = true;
+                        redirectTime = indResTime.redirectEnd - indResTime.redirectStart;
+                        dnstime = indResTime.domainLookupEnd - indResTime.domainLookupStart;
+                        tcpHandshakeTime = indResTime.connectEnd - indResTime.connectStart;
+                        secureConntime = (indResTime.secureConnectionStart > 0) ? (indResTime.connectEnd - indResTime.secureConnectionStart) : "0";
+                        responseTime = indResTime.responseEnd - indResTime.responseStart;
+                        fetchStToRespEndTime = (indResTime.fetchStart > 0) ? (indResTime.responseEnd - indResTime.fetchStart) : "0";
+                        reqStToRespEndTime = (indResTime.requestStart > 0) ? (indResTime.responseEnd - indResTime.requestStart) : "0";
+                        stToRespEnd = (indResTime.startTime > 0) ? (indResTime.responseEnd - indResTime.startTime) : "0";
+                        break;
+                    } else {
+                        isResourceDataAvailable = false;
+                    }
+                }
+            }
+            if (isResourceDataAvailable === false) {
+                redirectTime = -1;
+                dnstime = -1;
+                tcpHandshakeTime = -1;
+                secureConntime = -1;
+                responseTime = -1;
+                fetchStToRespEndTime = -1;
+                reqStToRespEndTime = -1;
+                stToRespEnd = -1;
+            }
+            let clientDuration;
+            if (spReqDuration == null) {
+                    clientDuration = -1;
+            } else {
+                    clientDuration = parseInt(spReqDuration, 10) - responseTime;
+            }
             event.end({
                 trees: content.trees?.length ?? 0,
                 blobs: content.blobs?.length ?? 0,
                 ops: content.ops?.length ?? 0,
                 headers: Object.keys(headers).length !== 0 ? true : undefined,
                 sprequestguid: response.headers.get("sprequestguid"),
                 sprequestduration: TelemetryLogger.numberFromString(response.headers.get("sprequestduration")),
+                isOptionsCall: isOptionsCall ? true : false,
+                clientDuration: TelemetryLogger.numberFromString(clientDuration),
```

Comment:
These should be numbers already, don't need to do numberFromString.

Reply:
Modified.